### ISSUE TITLE
Gnome HIG support?

### ISSUE THREAD

**Issue by ripefig** [phase1geo/Minder#144]
It would be nice if this app conformed a little better to Gnome HIG, when it's run on Gnome. There's another app called Drawing that has different layouts depending on which desktop it's running under (traditional, Gnome and Elementary)

**Comment by phase1geo** [comment]
I would be happy to review any pull requests which help Minder to better conform to the Gnome HIG and get those changes applied; however, the design goals that I am following with Minder conform to the elementary OS HIG which is the platform I have personal interest in.

**Comment by tronta** [comment]
@phase1geo, I would be happy to invest some time and come up with some UI proposals I'd like to show to you and get your feedback.
Would this be ok for you? If yes, what would be the best approach? Putting it simply in this ticket?

**Comment by phase1geo** [comment]
I'm thinking that this would be best handled in a GitHub discussion. I don't want to commit to making any layout changes without first understanding, conversing, and agreeing to a plan.

**Comment by tronta** [comment]
I totally agree, it makes no sense to just do something without having a common understanding and goal.
So will try to sit down the next days and write some ideas together and put it there. Never used GitHub discussion before, but I'll get it.

**Comment by tronta** [comment]
@phase1geo : not sure if you have seen it, but I have started a discussion [here](https://github.com/phase1geo/Minder/discussions/384).
Not sure if this is the right tool. I guess the wiki might be a better place.

To you think this approach is helpful? Should I continue?

**Comment by phase1geo** [comment]
The discussions board should be a good place to have these discussions.  I like to use the wiki for documentation rather than for discussions.

**Comment by tronta** [comment]
It would be a kind of concept document, which should be modifiable/commentable by multiple people. In discussion, I guess I'm the only one who can modify the document, right?

**Comment by tronta** [comment]
@phase1geo did you have a chance to look at the current state of my proposal?

**Comment by phase1geo** [comment]
I have read the portion of the proposal related to the header bar.

**Comment by QWxleA** [comment]
Really enjoy using Minder. On Gnome, though, some things are a bit off ;-) The concept document is rather grand, but as a simple Gnome user the following would be enough for me, to have Minder "fit" with the rest:

- Use [UI Icons - GNOME User Interface Guidelines documentation](https://developer.gnome.org/hig/guidelines/ui-icons.html)
- Use [Standard Keyboard Shortcuts - GNOME User Interface Guidelines documentation](https://developer.gnome.org/hig/reference/keyboard.html)

I don't really understand when vala builds what, but some not working code would be:

```vala
            if (Environment.get_variable ("XDG_CURRENT_DESKTOP") == "pantheon") {
                /* Adds keyboard shortcuts for the menu actions */
  private void add_keyboard_shortcuts( Gtk.Application app ) {

    app.set_accels_for_action( "win.action_save",          { "<Control>s" } );
    app.set_accels_for_action( "win.action_quit",          { "<Control>q" } );
    app.set_accels_for_action( "win.action_zoom_actual",   { "<Control>0" } );
    app.set_accels_for_action( "win.action_zoom_fit",      { "<Control>1" } );
    app.set_accels_for_action( "win.action_zoom_in",       { "<Control>plus" } );
    app.set_accels_for_action( "win.action_zoom_in",       { "<Control>equal" } );
    app.set_accels_for_action( "win.action_zoom_out",      { "<Control>minus" } );
    app.set_accels_for_action( "win.action_export",        { "<Control>e" } );
    app.set_accels_for_action( "win.action_print",         { "<Control>p" } );
    app.set_accels_for_action( "win.action_prefs",         { "<Control>comma" } );
    app.set_accels_for_action( "win.action_shortcuts",     { "F1" } );
    app.set_accels_for_action( "win.action_show_current",  { "<Control>6" } );
    app.set_accels_for_action( "win.action_show_style",    { "<Control>7" } );
    app.set_accels_for_action( "win.action_show_stickers", { "<Control>8" } );
    app.set_accels_for_action( "win.action_show_map",      { "<Control>9" } );
    app.set_accels_for_action( "win.action_next_tab",      { "<Control>Tab" } );
    app.set_accels_for_action( "win.action_prev_tab",      { "<Control><Shift>Tab" } );
  }
            } else  {
                gnome keybindings, like:
    app.set_accels_for_action( "win.action_shortcuts",     { "<control>?" } );
            }
```

(Copied from [MainWindow.vala](https://github.com/phase1geo/Minder/blob/e7e561e146b67ceaca3d6b4c915392e4db6a5b5c/src/MainWindow.vala#L479) and [midori](https://github.com/midori-browser/core/blob/d8546ca689af185a456ec1f394bb89926d7c142a/core/browser.vala#L346))

Would this be possible? Have the keybindings and icons 'set' during runtime, and not during compilation?

**Comment by phase1geo** [comment]
I can probably just add the Gnome shortcuts, but I would continue to do this at compile time. I don't want to load shortcuts from somewhere on the file system.

**Comment by QWxleA** [comment]
> I can probably just add the Gnome shortcuts, but I would continue to do this at compile time. I don't want to load shortcuts from somewhere on the file system.
No shortcuts from an external file sounds a bit yucky. 

Another solution would be a build switch in Meson to differentiate building for the Elementary app-store or building for flathup.org Would you agree to that? If you could add the if / then logic, I could send a pull request for the Gnome keybindings and icon settings.

**Comment by phase1geo** [comment]
How would meson know which AppStore it is being built for?

**Comment by QWxleA** [comment]
Looking into it, meson might be the wrong place, if you build flatpaks with app:
`BUILDFOR="gnome"  sudo flatpak-builder --install --force-clean ../build-minder com.github.phase1geo.minder.yml`

and then in your vala code: 
`if (Environment.get_variable ("BUILDFOR") == "gnome") {`

might work (sorry to make work for an option you yourself might not even need)

**Comment by phase1geo** [comment]
I have modified the keyboard shortcuts to conform the the Gnome HIG.  This is available in the 1.13.1 release of Minder.

As for the UI change requests, I have decided to not add support for this in the foreseeable future.
